### PR TITLE
Add information on NEST and multiprocessing

### LINKED ISSUE #1581: Documentation on nest and multi-processing / multi-threading
**Describe the flaw**
The simulator is built for parallel processing, but having many nodes in a cluster it is tempting to try to use python's multiprocessing package to speed up also other parts of the simulation scirpts. Information on the possible scenarios, relation to Python's multi-processing API, or multi-threading with other communications libraries is missing.

**Suggest an improvement**
References to a conceptual document describing the mpi/threading environment should be in relevant sections of guides and reference docs.

**Supporting materials**
Was [discussed](https://www.nest-simulator.org/mailinglist/hyperkitty/list/users@nest-simulator.org/thread/ZAXHALXHJTUOOX4VGD3T2FIMWNNNTILE/#ZAXHALXHJTUOOX4VGD3T2FIMWNNNTILE) on [NEST mailing list](https://nest-simulator.org/mailinglist)

**Desktop/Environment**
relevant for all users of clusters or multi-core hardware.

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR fixes #1581. 

It recommends against the use of Python's multiprocessing package on top of NEST's internal parallelization. I have also referenced [Ippen et al. (2017)](https://www.frontiersin.org/articles/10.3389/fninf.2017.00030/full), as suggested by @suku248. 

The full discussion can be found in the [mailing list](https://www.nest-simulator.org/mailinglist/hyperkitty/list/users@nest-simulator.org/thread/ZAXHALXHJTUOOX4VGD3T2FIMWNNNTILE/#ZAXHALXHJTUOOX4VGD3T2FIMWNNNTILE).

Perhaps we can update the section accordingly once we have a careful assessment about the interactions between NEST and multiprocessing?

### CODE REVIEW

File: doc/guides/parallel_computing.rst [hunk 1]
```diff
@@ -179,6 +179,14 @@ on your machine.
  can yield 20-30% improvement in simulation speed. Finding the optimal thread number for a
  specific situation might require a bit of experimenting.
 
+Multiprocessing
+----------------
+
+**Please do not use Python's multiprocessing package with NEST!**
+
+NEST internally parallelizes network construction [1]_ and maintains internal data structures in this process. Running
+several ``ConnectLayers()`` calls simultaneously can lead to unpredictable results.
```
Comment (terhorstd) [inline doc/guides/parallel_computing.rst:188]:
```suggestion
for example several ``Connect()`` calls simultaneously can interfere with the internal
parallelization and will likely lead to unpredictable/wrong results.
```

**Comment by terhorstd** [comment]
@suku248, @diesmann do you still want to comment on these changes?

File: doc/guides/parallel_computing.rst [hunk 2]
```diff
@@ -179,6 +179,15 @@ on your machine.
  can yield 20-30% improvement in simulation speed. Finding the optimal thread number for a
  specific situation might require a bit of experimenting.
 
+Multiprocessing
+----------------
+
+**Using Python's ``multiprocessing`` package with NEST may lead to unpredictable results!**
```
Comment (jougs) [inline doc/guides/parallel_computing.rst:185]:
```suggestion
**Using Python's ``multiprocessing`` module with NEST may lead to unpredictable results!**
```

**Review by jougs — CHANGES_REQUESTED** [review]
I just have a small suggestion, but am fine otherwise. Many thanks!

**Comment by sarakonradi** [comment]
Done!

**Review by terhorstd — APPROVED** [review]
LGTM

**Review by jougs — APPROVED** [review]

**Pull request merged**